### PR TITLE
use `isarray` for old browsers

### PR DESCRIPTION
**Pull request by vvo**
IE<=9 has no Array.isArray.

It's the only change needed to make this module work in IE8.

### CODE REVIEW

**Comment by vvo** [comment]
ping @ljharb

**Comment by ljharb** [comment]
Alternatively you could use the `es5-shim`, which would provide it. I can't imagine working in IE 8 without that :-)

**Comment by ljharb** [comment]
While I think this is great, I think that a dependency would break `component` usage. Can you make an internal `isArray` function instead?

**Comment by vvo** [comment]
@ljharb Yeah I know es5-shim is a great solution! It's just that when you find yourself authoring modules that need to be compatible with old browsers, you cannot just include es5-shim in your module.

Do you

**Comment by ljharb** [comment]
Unfortunately I don't use `component` - only `npm` and `browserify` - so I'm not sure what will break it, but the decision to support it was made before my arrival.

If you can verify that adding it as a `component` dep works, then yay! Otherwise an internal function would be preferred.

Thanks!

**Comment by vvo** [comment]
ok @ljharb I inlined isarray, could not make component install work.

File: index.js [hunk 1]
```diff
@@ -2,6 +2,11 @@ var hasOwn = Object.prototype.hasOwnProperty;
 var toString = Object.prototype.toString;
 var undefined;
 
+// inlined from https://github.com/juliangruber/isarray
+var isArray = Array.isArray || function (arr) {
+  return Object.prototype.toString.call(arr) == '[object Array]';
```
Comment (ljharb) [inline index.js]: Please use `===` here instead of `==`.

Comment (vvo) [inline index.js]: Solved @ljharb sorry for making this long for you :)

**Comment by ljharb** [comment]
One problem is that this will break code coverage - you'd need to have a test that deletes `Array.isArray`, makes sure it `extend` still works, and then restores it.

**Comment by vvo** [comment]
> One problem is that this will break code coverage - you'd need to have a test that deletes Array.isArray, makes sure it extend still works, and then restores it.

will do tonight

**Comment by vvo** [comment]
@ljharb fixed

File: index.js [hunk 2]
```diff
@@ -2,6 +2,15 @@ var hasOwn = Object.prototype.hasOwnProperty;
 var toString = Object.prototype.toString;
 var undefined;
 
+// inlined from https://github.com/juliangruber/isarray
+var isArray = function (arr) {
+	if (Array.isArray) {
+		return Array.isArray.call(null, arr);
```
Comment (ljharb) [inline index.js]: `Array.isArray(arr)` should be sufficient here, rather than `.call`.

**Comment by ljharb** [comment]
Thanks, this looks great!

File: index.js (hunk 2 above)
Comment (vvo) [inline index.js]: oh yes, was just confused here

**Pull request merged**